### PR TITLE
Fix responsive prop, add a warning when it's not affecting the transformation

### PR DESCRIPTION
**Pull request by maoznir**
This is a fix for https://github.com/cloudinary/cloudinary-react/issues/175

### CODE REVIEW

File: e2e-test/cypress/integration/responsive.spec.js [hunk 1]
```diff
@@ -4,6 +4,12 @@ describe('Responsive Image', () => {
     cy.visit('/');
     cy.get('#responsiveBtn').click();          // Click on button
   });
+  it('Responsive Image Override', () => {
+    cy.get('#responsive-override')
+      .should('have.attr', 'data-src').should('equal','http://res.cloudinary.com/demo/image/upload/c_scale,w_100/sample')
+    cy.get('#responsive-override')
+      .should('have.attr', 'src').should('equal','http://res.cloudinary.com/demo/image/upload/c_scale,w_100/sample')
```
Comment (asisayag2) [inline e2e-test/cypress/integration/responsive.spec.js:11]: really, it should have both?

File: e2e-test/cypress/integration/responsive.spec.js [hunk 2]
```diff
@@ -4,6 +4,12 @@ describe('Responsive Image', () => {
     cy.visit('/');
     cy.get('#responsiveBtn').click();          // Click on button
   });
+  it('Responsive Image Override', () => {
```
Comment (asisayag2) [inline e2e-test/cypress/integration/responsive.spec.js:7]: Can you explain what you mean by "responsive override"?

File: src/components/Image/Image.js [hunk 3]
```diff
@@ -19,7 +19,20 @@ class Image extends CloudinaryComponent {
    * @return true when this image element should be made responsive, false otherwise.
    */
   isResponsive = () => {
-    return this.props.responsive && this.imgElement && this.imgElement.current;
+    const {imgElement} = this;
```
Comment (asisayag2) [inline src/components/Image/Image.js:22]:
```suggestion
    const {responsive, width} = this.getExtendedProps();
    if (responsive && width !== 'auto') {
        console.warn(WARNING_CONST);
    }
    
    return responsive && this.imgElement && this.imgElement.current;
```

File: src/components/Image/Image.js [hunk 4]
```diff
@@ -43,34 +56,43 @@ class Image extends CloudinaryComponent {
...
 
+    // Set dataSrc if lazy loading and not in view
+    if (!isInView && this.shouldLazyLoad(options)) {
+      attributes.dataSrc = getConfiguredCloudinary(options).url(options.publicId, options);
+      delete attributes.src;
+    }
+
+    if (attributes.dataSrc) {
+      attributes['data-src'] = attributes.dataSrc;
+    }
```
Comment (asisayag2) [inline src/components/Image/Image.js:74]: ???

File: e2e-test/cypress/integration/responsive.spec.js (hunk 2 above)
Comment (maoznir) [reply e2e-test/cypress/integration/responsive.spec.js:7]: Added comments

File: e2e-test/cypress/integration/responsive.spec.js (hunk 1 above)
Comment (maoznir) [reply e2e-test/cypress/integration/responsive.spec.js:11]: Yes, added comments to explain

File: src/components/Image/Image.js (hunk 3 above)
Comment (maoznir) [reply src/components/Image/Image.js:22]: thanks, refactored as suggested

File: src/components/Image/Image.js [hunk 5]
```diff
@@ -43,34 +56,43 @@ class Image extends CloudinaryComponent {
...
 
     let attributes = {...getImageTag(options).attributes(), ...nonCloudinaryProps};
+    attributes = Util.withCamelCaseKeys(attributes);
 
     // Set placeholder Id
     if (placeholder && attributes.id) {
       attributes.id = attributes.id + '-cld-placeholder';
     }
 
+    // Set dataSrc if lazy loading and not in view
```
[the remark this answers is not in the document]
Comment (maoznir) [reply src/components/Image/Image.js:66]: I needed to update this because i added withCamelCaseKeys().

File: src/components/Image/Image.js (hunk 4 above)
Comment (maoznir) [reply src/components/Image/Image.js:74]: Looking at this again, calculating a new src isn't required, i've updated this part, please see updated commit

**Pull request merged**